packages/transformers/js/core/src/env_replacer.rs

```diff
@@ -41,6 +41,32 @@ impl<'a> Fold for EnvReplacer<'a> {
       }
     }
 
+    if let Expr::Bin(ref binary) = node {
+      if let Expr::Member(ref member) = &*binary.right {
+        if let Expr::Lit(Lit::Str(ref left)) = &*binary.left {
+          if let Expr::Ident(ref ident) = &*member.obj {
+            if &ident.sym == "process" {
```

Comment:
This also matches if there is a local variable called `process`.
So you need to do something along the lines of `if !decls.contains(ident)`. You find other places places where this is done by searching for `decls.contains`.

Reply:
I see. sorry, I did not notice this. Indeed, this process is done in `typeof_replacer`. [c8ba4ba](https://github.com/parcel-bundler/parcel/pull/7954/commits/c8ba4baf8e61237d2aba6cccfdfba1de382da455) 👍